desktop/core/src/desktop/api2_tests.py

```diff
@@ -699,6 +699,31 @@ def test_create(self):
         json.loads(Document2.objects.get(type='gist', uuid=gist2['uuid']).data)['statement_raw']
     )
 
+  def test_handle_multiple_gist_dir(self):
+    # Create multiple gist directories and then make a new gist
+    home_dir = Directory.objects.get_home_directory(self.user)
+
+    gist_dir1 = Directory.objects.create(name=Document2.GIST_DIR, owner=self.user, parent_directory=home_dir)
+    gist_dir2 = Directory.objects.create(name=Document2.GIST_DIR, owner=self.user, parent_directory=home_dir)
+
+    assert_equal(2, Directory.objects.filter(name=Document2.GIST_DIR, type='directory').count())
+
+    response = self._create_gist(
+        statement='SELECT 12345',
+        doc_type='hive-query',
+        name='test_gist_create',
+    )
+    gist = json.loads(response.content)
+
+    # Gist directories merged into one
```

Comment:
Confusing comment as AFAIK this is happening in create_gist (so line 710 better)